src/main/java/core/basesyntax/operation/ReturnHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class ReturnHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.RETURN) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopServiceImpl.addFruits(fruit, quantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for ReturnHandler");

```

Comment:
As mentioned in the previous comment, the operation type check should be removed. Therefore, this `RuntimeException` is not needed and can be removed.